src/main/java/com/google/api/codegen/configgen/transformer/DiscoConfigTransformer.java

```diff
@@ -100,42 +118,69 @@ private LicenseView generateLicense() {
       interfaceView.collections(collectionTransformer.generateCollections(collectionNameMap));
       interfaceView.methods(
           methodTransformer.generateMethods(
-              new DiscoInterfaceModel(resource.getKey(), model), collectionNameMap));
+              new DiscoInterfaceModel(resource, model), collectionNameMap));
       interfaces.add(interfaceView.build());
     }
     return interfaces.build();
   }
 
   /**
+   * Get the resource name for a method. Qualifies the resource name if it clashes with another
+   * resource with the same name but different canonical path.
+   */
+  private String getResourceIdentifier(
+      Method method, String parentName, SetMultimap<String, String> resourceToNamePatternMap) {
+    String resourceName = DiscoGapicNamer.getResourceIdentifier(method.flatPath()).toLowerCamel();
+    if (resourceToNamePatternMap.get(resourceName).size() == 1) {
+      return resourceName;
+    } else {
+      // Qualify resource name to avoid naming clashes with other methods with same name pattern.
+      return DiscoGapicNamer.getQualifiedResourceIdentifier(method, parentName).toLowerCamel();
+    }
+  }
+
+  /**
    * Examines all of the resource paths used by the methods, and returns a map from each unique
-   * resource paths to a short name used by the collection configuration.
+   * canonical resource path to a resource identifier (a short name used by the collection
+   * configuration). Each resource path is merely a string describing the fields in the entity, and
+   * the resource path might not be the same as the RPC endpoint URI. The resource identifier is
+   * globally unique within each API. Many methods may use the same resource collection. The
+   * resource identifier will be qualified (with the name of a parent resource) iff there are two
+   * resource identifiers with different canonical resource paths.
```

Comment:
Shouldn't this be: "iff there are two *or more* resource identifiers..."?

Reply:
Yes :) Done.